From 9872cc3a8d25e55cd164c89b6069dc7d79e54b25 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Fri, 8 Feb 2019 13:28:26 -0800
Subject: [PATCH] TRITON-1026 CNAPI CommandExecute should not use Ur (remove
 doc section for removed Ur-only CommandExecute endpoint) Reviewed by: Josh
 Wilsdon <josh@wilsdon.ca> Approved by: Josh Wilsdon <josh@wilsdon.ca>

---
 docs/index.md           | 25 +------------------------
 docs/index/index.md.ejs |  2 +-
 lib/endpoints/ur.js     | 17 +++--------------
 3 files changed, 5 insertions(+), 39 deletions(-)

diff --git a/docs/index.md b/docs/index.md
index b8a38bb..d5ea770 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -1,6 +1,6 @@
 ---
 title: CNAPI (Compute Node API) Design
-apisections: Allocation API, Boot Parameters API, Compute Node Agent Tasks API, Miscellaneous API, Remote Execution API, Remote Execution API (deprecated), Server API, Virtual Machine API, Virtual Machine Images API, Virtual Machine Snapshots API, Waitlist API, ZFS API (deprecated)
+apisections: Allocation API, Boot Parameters API, Compute Node Agent Tasks API, Miscellaneous API, Remote Execution API (deprecated), Server API, Virtual Machine API, Virtual Machine Images API, Virtual Machine Snapshots API, Waitlist API, ZFS API (deprecated)
 markdown2extras: tables, code-friendly
 ---
 <!--
@@ -1101,29 +1101,6 @@ None.
 
 
 
-# Remote Execution API
-
-## CommandExecute (POST /servers/:server_uuid/execute)
-
-Synchronously execute a command on the target server.
-
-### Inputs
-
-| Param  | Type   | Description                                             |
-| ------ | ------ | ------------------------------------------------------- |
-| args   | Array  | Array containing arguments to be passed in to command   |
-| env    | Object | Object containing environment variables to be passed in |
-| script | String | Script to be executed. Must have a shebang line         |
-
-
-### Responses
-
-| Code | Type | Description    |
-| ---- | ---- | -------------- |
-| 404  | None | No such server |
-
-
-
 # Remote Execution API (deprecated)
 
 ## CommandExecute (deprecated) (POST /servers/:server_uuid/execute)
diff --git a/docs/index/index.md.ejs b/docs/index/index.md.ejs
index bf4b8a4..d0fb982 100644
--- a/docs/index/index.md.ejs
+++ b/docs/index/index.md.ejs
@@ -10,7 +10,7 @@ markdown2extras: tables, code-friendly
 -->
 
 <!--
-    Copyright (c) 2018, Joyent, Inc.
+    Copyright (c) 2019, Joyent, Inc.
 -->
 
 <!--
diff --git a/lib/endpoints/ur.js b/lib/endpoints/ur.js
index 80861f9..8a7e1df 100644
--- a/lib/endpoints/ur.js
+++ b/lib/endpoints/ur.js
@@ -5,11 +5,12 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
- * ur.js: Ur endpoints
+ * ur.js: Ur support for the fallback handling of the CommandExecute
+ * endpoint (defined in lib/endpoints/server.js).
  *
  * The /servers/:server_uuid/execute endpoint invokes scripts on servers. If the
  * server has old agents that don't support the command_execute task, the
@@ -30,19 +31,7 @@ function Ur() {}
 
 /**
  * Synchronously execute a command on the target server.
- *
- * @name CommandExecute
- * @endpoint POST /servers/:server_uuid/execute
- * @section Remote Execution API
- *
- * @param {Array} args Array containing arguments to be passed in to command
- * @param {Object} env Object containing environment variables to be passed in
- * @param {String} script Script to be executed. Must have a shebang line
- *
- * @response 404 None No such server
- * @resposne 500 None Error occurred executing script
  */
-
 Ur.execute = function handlerUrExecute(req, res, next) {
     var rules = {
         'args': ['optional', 'isArrayType'],
-- 
2.21.0

